keps/sig-instrumentation/3498-extending-stability/README.md

```diff
@@ -0,0 +1,179 @@
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The metric stability framework was originally introduced with the intent of safeguarding significant metrics from being broken downstream. Metrics could be deemed `stable` or `alpha`, and only `stable` metrics would have stability guarantees. 
+
+This KEP intends to propose additional stability classes to extend on the existing metrics stability framework, such that we can achieve parity with the various stages of the feature release cycle.
+
+## Motivation
+
+It's become more obvious recently that we need additional stability classes, particularly in respect to various stages of feature releases. This has become more obvious with the advent of PRR (production readiness reviews) and [mandated production readiness metrics](https://github.com/kubernetes/community/blob/cf715a4404c4cefcfb52278bc128b7a765373fc7/sig-architecture/production-readiness.md#common-feedback-from-reviewers). 
+
+### Goals
+
+Introduce two more metric classes: `beta`, corresponding to the `beta` stage of feature release, and `internal` which corresponds to internal development related metrics. 
+
+
+### Non-Goals
+
+- establishing if specific metrics fall into a stability class, this exercise is left for component owners, who own their own metrics
+- establishing the guarantees of the `beta` stability class, this will be an exercise we will defer until the beta version of this KEP.
+
+
+## Proposal
+
+We're proposing adding additional metadata fields to Kubernetes metrics. Specifically we want to add the following stability levels:
+
+- `Internal` - representing internal usages of metrics (i.e. classes of metrics which do not correspond to features)
```

Comment:
Is the requirement really "not correspond to features"?

What I had on my mind was that e.g. some feature metrics may be so low-level that a typical operator won't understand them (or won't be able to react to them properly), so I would consider them also "internal" metrics.

Isn't this the case?

Reply:
Sure, amended.